staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal.go

```diff
@@ -59,6 +87,39 @@ func (s *JSONSchemaPropsOrBool) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+func (s JSONSchemaPropsOrBool) MarshalCBOR() ([]byte, error) {
```

Comment:
I'm curious what the behavior of the API would have been prior to this PR if the CBOR feature was enabled and CBOR was requested from the CRD API

Reply:
I added a new section to the KEP discussing situations where it is asked to encode or decode into an object that implements json.Marshaler, json.Unmarshaler, encoding.TextMarshaler, and/or encoding.TextUnmarshaler: https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/4222-cbor-serializer/README.md#custom-json-marshalers.

The short version is that, for now, the CBOR serializer returns an error refusing to touch it. As a beta criteria, it will call the implemented custom marshaler/unmarshaler and then transcode the result to CBOR.